commcare_connect/opportunity/tasks.py

```diff
@@ -227,3 +231,31 @@ def send_sms_task(user_ids: list[int], body: str):
     user_phone_numbers = User.objects.filter(id__in=user_ids).values_list("phone_number", flat=True)
     for phone_number in user_phone_numbers:
         send_sms(phone_number, body)
+
+
+@celery_app.task()
+def download_user_visit_attachments(user_visit_id: id):
+    user_visit = UserVisit.objects.get(id=user_visit_id)
+    api_key = user_visit.opportunity.api_key
+    blobs = user_visit.form_json.get("attachments", {})
+    domain = user_visit.opportunity.deliver_app.cc_domain
+    form_id = user_visit.xform_id
+    for name, blob in blobs.items():
+        if name == "form.xml":
+            continue
+        url = f"{settings.COMMCARE_HQ_URL}/a/{domain}/api/form/attachment/{user_visit.xform_id}/{name}"
+
+        blob_meta = BlobMeta(
+            name=name, parent_id=form_id, content_length=blob["length"], content_type=blob["content_type"]
+        )
+        with transaction.atomic():
+            try:
+                blob_meta.save()
+            except IntegrityError:
+                # attachment already exists
+                continue
+            response = httpx.get(
+                url,
+                headers={"Authorization": f"ApiKey {api_key.user.email}:{api_key.api_key}"},
+            )
+            default_storage.save(blob_meta.blob_id, ContentFile(response.content, name))
```

Comment:
There was an error on staging on this line due to the `blob_meta.blob_id` being a UUID object. That needs to be converted to a string representation before the file can be saved.

Reply:
This probably is due to the difference being the local storage vs S3, though I wouldn't expect the implementation to differ. I have casted it to str now.